pkg/install/helm/cli.go

```diff
@@ -55,6 +55,47 @@ func NewCLI(binary string, kubeconfig string, kubeContext string, timeout time.D
 	}, nil
 }
 
+func (c *cli) BuildChartDependencies(chartDirectory string, flags []string) (err error) {
+	command := []string{
+		"dependency",
+		"build",
+	}
+
+	chart, err := LoadChart(chartDirectory)
+	if err != nil {
+		return err
+	}
+
+	for idx, dep := range chart.Dependencies {
+		repoAddFlags := []string{
+			"repo",
+			"add",
+			fmt.Sprintf("dep-%s-%d", chart.Name, idx),
+			dep.Repository,
+		}
+
+		repoRemoveFlags := []string{
+			"repo",
+			"remove",
+			fmt.Sprintf("dep-%s-%d", chart.Name, idx),
+		}
+
+		defer func() {
+			_, removeErr := c.run("default", repoRemoveFlags...)
+			if err != nil {
+				err = removeErr
+			}
+		}()
+
+		_, err = c.run("default", repoAddFlags...)
```

Comment:
The defer should only happen after the `repo add` already succeeded, no?

Also, in case of an error, why not `break` out of the loop and skip needlessly adding and removing more repos? :)

Reply:
1) I'd rather have the removal "scheduled" before I add the repo, since it can be "added" and then fail during update, which would lead to a repo created in helm anyway, just not working correctly. `repo remove` cleans it up anyway.

2) yeah, I will do that :)